irksome/explicit_stepper.py

```diff
@@ -20,3 +20,16 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
             F, butcher_tableau, t, dt, u0, bcs=bcs,
             solver_parameters=solver_parameters, appctx=appctx,
             nullspace=None)
+
+    # DAE treatment of the boundary conditions in this case says that
+    # we should impose the BCs so that they are satisfied for the next
+    # stage for all but that last stage, and that they are satisfied
+    # for the next timestep for the last stage.
+    def update_bc_constants(self, AAb, CCone, i, a_vals, d_val, c):
```

Comment:
Please explain the +1 shift in more detail in the docstring

Reply:
I wrote a little more math in the docstring to explain this.  Is that enough?